spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/Commons.java

```diff
@@ -105,12 +105,13 @@ public static void assertReloadLogStatements(String left, String right, String a
 					.getStdout();
 			await().pollInterval(Duration.ofSeconds(5)).atMost(Duration.ofSeconds(180)).until(() -> {
 
-				String allLogs = CONTAINER.execInContainer("kubectl", "logs", appPodName.trim()).getStdout();
-				LOG.info("==========================================================================================");
-				LOG.info(allLogs);
-				LOG.info("==========================================================================================");
-				if (allLogs.contains(left)) {
-					Assertions.assertFalse(allLogs.contains(right));
+				String present = CONTAINER
```

Comment:
slightly re-write the way to get the logs

Reply:
I decided to use `grep` here because this was still sometimes failing for me. tbh, I still don't know why and I tried many things here. My hint is that in the previous version, I was retrieving all the logs and was doing a `contains`. Since there are many of them, and they are extracted once in 3 seconds, it sometimes breaks something, don't know what. Instead of digging into test-containers very deep, I decided to grep here. It's easier and a lot faster